src/components/post-card/PostCard.js

```diff
@@ -1,36 +1,58 @@
-import React from 'react';
+import React, {Component} from 'react';
 
 import { Comment } from '../comment/Comment';
 import DefaultImg from '../../assets/default-empty-img.png';
 import './PostCard.scss';
 
-// todo: переписать эту компоненту, чтоб она стала классовой
-//  в ней должно появиться свойство state
-//  в котором будет флажок showComments
-//  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments"
+// todo: переписать эту компоненту, чтоб она стала классовой +
+//  в ней должно появиться свойство state +
+//  в котором будет флажок showComments +
+//  и будет кнопка, лейба которой будет либо "show comments" либо "hide comments" +
 
-export function PostCard(props) {
-  const { post, hasImage, comments = [], author } = props;
+export class PostCard extends Component {
+  constructor(props) {
+    super(props);
 
-  console.log('hasImage', hasImage);
-  const { title, body } = post;
+    this.state = {
+      isOpen: false,
+      showComments: 'show comm'
+    }
+  }
 
-  const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+  onClick = () => {
+  console.log ('click');
+    this.state.isOpen ? this.setState( {
+          showComments: 'show comm'
+        }) : this.setState( {
+      showComments: 'hide comm'
+    });
+    this.setState( {
+      isOpen: !this.state.isOpen
+    })
+};
```

Comment:
Please try to simplify this code

try to use only 1 this.setState 
and make a checks for each property  that you set

also you can set several properties in 1 setState

````
this.setState({
isOpen: ....
showComments: .....
})

````